AutoCollection/AzureFunctionsHook.ts

```diff
@@ -34,84 +43,110 @@ export class AzureFunctionsHook {
-                extractedContext.customProperties.setProperty("LogLevel", ctx.traceContext.attributes["LogLevel"]);
-                extractedContext.customProperties.setProperty("Category", ctx.traceContext.attributes["Category"]);
-                extractedContext.customProperties.setProperty("HostInstanceId", ctx.traceContext.attributes["HostInstanceId"]);
-                extractedContext.customProperties.setProperty("AzFuncLiveLogsSessionId", ctx.traceContext.attributes["#AzFuncLiveLogsSessionId"]);
-            }
-        }
-        catch (err) {
-            Logging.warn("Failed to propagate context in Azure Functions", err);
-            originalCallback(ctx, request);
-            return;
-        }
-        if (!extractedContext) {
-            // Correlation Context could be disabled causing this to be null
-            Logging.warn("Failed to create context in Azure Functions");
-            originalCallback(ctx, request);
-            return;
-        }
-
-        CorrelationContextManager.wrapCallback(async () => {
-            const startTime = Date.now(); // Start trackRequest timer
-            originalCallback(ctx, request);
-            try {
-                if (this._autoGenerateIncomingRequests) {
-                    let statusCode = 200; //Default
-                    if (ctx.res) {
-                        if (ctx.res.statusCode) {
-                            statusCode = ctx.res.statusCode;
-                        }
-                        else if (ctx.res.status) {
-                            statusCode = ctx.res.status;
+    private _addPostInvocationHook() {
+        if (!this._postInvocationHook) {
+            this._postInvocationHook = this._functionsCoreModule.registerHook("postInvocation", async (postInvocationContext: PostInvocationContext) => {
+                try {
+                    if (this._autoGenerateIncomingRequests) {
+                        const ctx: Context = <Context>postInvocationContext.invocationContext;
+                        if (this._isHttpTrigger(ctx)) {
+                            const request: HttpRequest = postInvocationContext.inputs[0];
+                            if (request) {
+                                const startTime: number = postInvocationContext.hookData.appInsightsStartTime || Date.now();
+                                const extractedContext: CorrelationContext | undefined = postInvocationContext.hookData.appInsightsExtractedContext;
+                                if (!extractedContext) {
+                                    this._createIncomingRequestTelemetry(ctx, request, startTime, null);
+                                }
+                                CorrelationContextManager.runWithContext(extractedContext, () => {
+                                    this._createIncomingRequestTelemetry(ctx, request, startTime, extractedContext.operation.parentId);
```

Comment:
Can you clarify why `_createIncomingRequestTelemetry` is called twice one right after the other? Maybe the second one is supposed to be in an `else` case?